From a2ce09d7fed1f6fed8d68b5eba585cee452eac04 Mon Sep 17 00:00:00 2001
From: Janusz Dziedzic <januszx.dziedzic@intel.com>
Date: Thu, 12 May 2016 08:53:05 +0200
Subject: [PATCH 57/78] Revert "usb: dwc3: Add disable SUSPHY in DCTL and move
 link to U0"

This reverts commit a643221927704ad5a85ac668b393ba9bac41069d.

Signed-off-by: Janusz Dziedzic <januszx.dziedzic@intel.com>

Conflicts:
	drivers/usb/dwc3/gadget.c

Community version will be used instead.

Change-Id: I9da82996b586e6a0b703473327cbfd109e70c783
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-19337
Reviewed-on: https://android.intel.com:443/511182
---
 drivers/usb/dwc3/gadget.c | 19 -------------------
 1 file changed, 19 deletions(-)

diff --git a/drivers/usb/dwc3/gadget.c b/drivers/usb/dwc3/gadget.c
index 045cdda..cb355eb 100644
--- a/drivers/usb/dwc3/gadget.c
+++ b/drivers/usb/dwc3/gadget.c
@@ -299,7 +299,6 @@ int dwc3_send_gadget_ep_cmd(struct dwc3 *dwc, unsigned ep,
 	struct dwc3_ep		*dep = dwc->eps[ep];
 	u32			timeout = 500;
 	u32			reg;
-	u32			phycfg_val;
 	int			ret = -EINVAL;
 
 	trace_dwc3_gadget_ep_cmd(dep, cmd, params);
@@ -308,18 +307,12 @@ int dwc3_send_gadget_ep_cmd(struct dwc3 *dwc, unsigned ep,
 	dwc3_writel(dwc->regs, DWC3_DEPCMDPAR1(ep), params->param1);
 	dwc3_writel(dwc->regs, DWC3_DEPCMDPAR2(ep), params->param2);
 
-	phycfg_val = dwc3_readl(dwc->regs, DWC3_GUSB2PHYCFG(0));
-
-	dwc3_writel(dwc->regs, DWC3_GUSB2PHYCFG(0),
-		(phycfg_val & ~DWC3_GUSB2PHYCFG_SUSPHY));
-
 	dwc3_writel(dwc->regs, DWC3_DEPCMD(ep), cmd | DWC3_DEPCMD_CMDACT);
 	do {
 		reg = dwc3_readl(dwc->regs, DWC3_DEPCMD(ep));
 		if (!(reg & DWC3_DEPCMD_CMDACT)) {
 			dwc3_trace(trace_dwc3_gadget, "Command Complete --> %d",
 					DWC3_DEPCMD_STATUS(reg));
-			dwc3_writel(dwc->regs, DWC3_GUSB2PHYCFG(0), phycfg_val);
 			if (DWC3_DEPCMD_STATUS(reg))
 				break;
 
@@ -950,7 +943,6 @@ static int __dwc3_gadget_kick_transfer(struct dwc3_ep *dep, u16 cmd_param,
 	struct dwc3			*dwc = dep->dwc;
 	int				ret;
 	u32				cmd;
-	u32				reg;
 
 	if (start_new && (dep->flags & DWC3_EP_BUSY)) {
 		dwc3_trace(trace_dwc3_gadget, "%s: endpoint busy", dep->name);
@@ -986,17 +978,6 @@ static int __dwc3_gadget_kick_transfer(struct dwc3_ep *dep, u16 cmd_param,
 		params.param0 = upper_32_bits(req->trb_dma);
 		params.param1 = lower_32_bits(req->trb_dma);
 		cmd = DWC3_DEPCMD_STARTTRANSFER;
-		if (dwc->gadget.speed != USB_SPEED_SUPER &&
-		  (dwc3_readl(dwc->regs, DWC3_DSTS)
-		  & DWC3_DSTS_USBLNKST_MASK) != 0) {
-			reg = dwc3_readl(dwc->regs, DWC3_DCTL);
-			reg &= ~(DWC3_DCTL_ULSTCHNGREQ_MASK);
-			dwc3_writel(dwc->regs, DWC3_DCTL, reg);
-			reg |= DWC3_DCTL_ULSTCHNG_RECOVERY;
-			dwc3_writel(dwc->regs, DWC3_DCTL, reg);
-			reg &= ~(DWC3_DCTL_ULSTCHNGREQ_MASK);
-			dwc3_writel(dwc->regs, DWC3_DCTL, reg);
-		}
 	} else {
 		cmd = DWC3_DEPCMD_UPDATETRANSFER;
 	}
-- 
1.9.1

